generator.go

```diff
@@ -59,16 +60,20 @@ func NewTraceGenerator(tsender Sender, getFielder func() *Fielder, log Logger, o
 // - nspans is the number of spans in a trace.
 // If nspans is less than depth, the trace will be truncated at nspans.
 // If nspans is greater than depth, some of the children will have siblings.
-func (s *TraceGenerator) generate_spans(ctx context.Context, fielder *Fielder, level int, depth int, nspans int, timeRemaining time.Duration) {
-	if depth == 0 || nspans == 0 {
+func (s *TraceGenerator) generate_spans(ctx context.Context, fielder *Fielder, level int, depth int, nspans int, timeRemaining time.Duration, debugCount *int) {
+	if nspans == 0 || timeRemaining <= 0 {
```

Comment:
Could this rewritten to always do the timeRemaining check first and sleep, if that doesn't happen, then we check if nspans == 0?